src/hoverifier.ts

```diff
@@ -687,7 +687,7 @@ export function createHoverifier<C extends object, D, A>({
                     },
                     end: {
                         line: hoverOrError.range.end.line + 1,
-                        character: hoverOrError.range.end.character + 1,
+                        character: hoverOrError.range.end.character,
```

Comment:
It definitely feels wrong to remove the `+ 1` from the character calculation but not from the line. That can't be right, and it will probably make the code more confusing/harder to debug for the next issue.

The comment above indicates that this transformation is done because the input LSP range is 0-indexed, but other parts of the code are 1-indexed.

```ts
                // The result is 0-indexed; the code view is treated as 1-indexed.
```

The issue here is probably more of a problem with the logic that maps the range to nodes treating the end as inclusive as opposed to exclusive, and the fix would be in that logic, as opposed to here. So something involving `getTokenAtPositionOrRange()` or `getTokenAtPositionOrRange()` probably.

BIG DISCLAIMER THAT I'M JUST GUESSING 😄

Reply:
@felixfbecker I agree, and I opened this PR to get exactly this type of advice. Who knows this code well enough to be able to formulate an actual solution?

Reply:
@tjkandala when he's back from vacation

Reply:
Bump @tjkandala

Reply:
He's still out of office today :)

Reply:
@tjkandala have a good vacay today cuz ur gettin pinged tmr

Reply:
@felixfbecker Looks like you're right (fix in PR #366)